Security/PasswordResetToken.php

```diff
@@ -58,4 +58,16 @@ public static function generate()
 
         return new self($token);
     }
+
+    /**
+     * Check if a token is equal to a different token.
+     *
+     * @param PasswordResetToken $token
+     *
+     * @return bool
+     */
+    public function equals(PasswordResetToken $token)
+    {
+        return $token->getToken() === $this->token;
```

Comment:
private fields kan je dus vanuit de class zelf aanroepen, ook als het een andere instantie is.